File: leetcode/problems/cpp/house-robber.cpp
// by @codeAbinash
// Time : O(n)
// Space : O(1)

#include "vector"
using namespace std;

class Solution {
  public:
   int rob(vector<int>& nums) {
      int rob1 = 0, rob2 = 0, tmp = 0;
      for (int i = 0; i < nums.size(); i++) {
         tmp = max(rob1 + nums[i], rob2);
         rob1 = rob2;
         rob2 = tmp;
      }
      return rob2;
   }
};

File: leetcode/problems/java/house-robber.java
class Solution {
  public int rob(int[] nums) {
    int rob1 = 0, rob2 = 0;
    for (int i = 0; i < nums.length; i++) {
      int temp = Math.max(rob1 + nums[i], rob2);
      rob1 = rob2;
      rob2 = temp;
    }
    return rob2;
  }
}